csrc/sys_utils.cpp

```diff
@@ -22,6 +22,10 @@
 #include <unistd.h>
 #include <cstdio>
 #include <cstdlib>
+#ifndef _GNU_SOURCE
+#define _GNU_SOURCE
+#endif
```

Comment:
Can you code-comment why?

Reply:
Will this cause incompatibility for any platforms we support or want to support? cc @xwang233

Reply:
I don't know what `_GNU_SOURCE` does

Reply:
Added a comment

Reply:
It's a GNU extension, so practically speaking, I'd be really surprised if there's any distribution that doesn't support it.